modules/core/shared/src/main/scala/gem/util/Format.scala

```diff
@@ -4,16 +4,28 @@
 package gem.util
 
 import cats.arrow._
+import cats.implicits._
 import monocle.{ Iso, Prism }
+import scala.reflect.runtime.universe.TypeTag
 
 /**
  * A normalizing optic, isomorphic to Prism but with different laws, specifically `getOption`
  * need not be injective; i.e., distinct inputs may have the same getOption result, which combined
  * with a subsequent `reverseGet` yield a normalized form for A. Composition with stronger optics
  * (`Prism` and `Iso`) yields another `Format`.
  */
+@SuppressWarnings(Array("org.wartremover.warts.Null"))
 final case class Format[A, B](getOption: A => Option[B], reverseGet: B => A) {
 
+  /** Like getOption, but throws IllegalArgumentException on failure. */
+  @SuppressWarnings(Array("org.wartremover.warts.DefaultArguments", "org.wartremover.warts.Throw"))
+  def unsafeGet(a: A)(implicit ev: TypeTag[B] = null): B =
```

Comment:
Can you tell a bit more about the default `null` argument? when would it be set?

Reply:
If there's a `TypeTag[B]` in implicit scope (which will be the case if `B` is a concrete type) then it will be passed, otherwise `null` will be passed. So this lets us say ok if `B` is instantiated to a concrete type then we can include that information in the error message, otherwise we just skip that part. This probably isn't a great pattern to use all over the place but for diagnostic information it can be helpful.

Reply:
When would `B` be not a concrete type?

Reply:
In a generic context like in `foo` in the example above.

```scala
def foo[A,B](f: Format[A,B], a: A) = f.unsafeGet(a) // no tag available here
```

Reply:
I see, but wouldn't it need to be a concrete type when called? or is the `TypeTag` decided at compilation time?

Reply:
Remember that implicit arguments are resolved statically at the call site. `foo` isn't asking for `(implicit ev: TypeTag[B])` so the only way to compile it is to pass `null`.

Reply:
So without the default `null` param it wouldn't compile?

Reply:
Correct. It would say it can't find an implicit `TypeTag[B]`.

Reply:
```
@ import reflect.runtime.universe.TypeTag 
import reflect.runtime.universe.TypeTag

@ implicitly[TypeTag[Int]] // ok 
res1: TypeTag[Int] = TypeTag[Int]

@ def foo[A] = implicitly[TypeTag[A]] // nope 
cmd2.sc:1: No TypeTag available for A
def foo[A] = implicitly[TypeTag[A]] // nope
                       ^
Compilation Failed

@ def foo[A: TypeTag] = implicitly[TypeTag[A]] // ok 
defined function foo
```

Reply:
That's pretty cool, thanks for the explanation.